backend/src/main/scala/cromwell/backend/WorkflowBackendActor.scala

```diff
@@ -0,0 +1,75 @@
+package cromwell.backend
+
+import akka.actor.{ActorRef, ActorLogging, Actor}
+import akka.event.LoggingReceive
+import cromwell.backend.WorkflowBackendActor._
+import cromwell.core.CallOutputs
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+import WorkflowBackendActor.ExecutionFailed
+
+object WorkflowBackendActor {
+
+  // Commands
+  sealed trait WorkflowBackendActorMessage
+
+  case class Execute(jobDescriptor: BackendJobDescriptor) extends WorkflowBackendActorMessage
+  case class Recover(jobDescriptor: BackendJobDescriptor) extends WorkflowBackendActorMessage
+  case class Abort(jobKey: BackendJobDescriptorKey) extends WorkflowBackendActorMessage
+
+  // Responses
+  sealed trait WorkflowBackendActorResponse
+
+  sealed trait ExecutionResponse extends WorkflowBackendActorResponse
+  case class ExecutionSucceeded(jobKey: BackendJobDescriptorKey, callOutputs: CallOutputs) extends ExecutionResponse
+  object ExecutionFailed {
+    def make(backendJobDescriptor: BackendJobDescriptor, throwable: Throwable) = ExecutionFailed(backendJobDescriptor.key, throwable)
+  }
+  case class ExecutionFailed(jobKey: BackendJobDescriptorKey, throwable: Throwable) extends ExecutionResponse
+  case class ExecutionFailedRetryable(jobKey: BackendJobDescriptorKey, throwable: Throwable) extends ExecutionResponse
+
+  sealed trait AbortResponse extends WorkflowBackendActorResponse
+  case class AbortSucceeded(jobKey: BackendJobDescriptorKey) extends AbortResponse
+  case class AbortFailed(jobKey: BackendJobDescriptorKey, throwable: Throwable) extends AbortResponse
+}
+
+/**
+  * Workflow-level actor for executing, recovering and aborting jobs.
+  */
+trait WorkflowBackendActor extends Actor with ActorLogging {
+
+  protected implicit def ec: ExecutionContext
+  protected def workflowDescriptor: BackendWorkflowDescriptor
+
+  def receive: Receive = LoggingReceive {
+    case Execute(jobDescriptor) => performActionThenRespond(execute, jobDescriptor, onFailure = ExecutionFailed.make)
+    case Recover(jobDescriptor) => performActionThenRespond(recover, jobDescriptor, onFailure = ExecutionFailed.make)
+    case Abort(jobKey)          => performActionThenRespond(abort, jobKey, onFailure = AbortFailed)
+  }
+
+  private def performActionThenRespond[A](operation: A => Future[WorkflowBackendActorResponse],
+                                          operationArgument: A,
+                                          onFailure: (A, Throwable) => WorkflowBackendActorResponse,
+                                          sndr: ActorRef = sender()) = {
```

Comment:
So you don't close over sender. Wouldn't be a huge deal  if we were good about not having multiple computations at once in an actor but that's how we roll.

That said I really dislike how it is done here, seems unintuitive and potentially problematic - I'm leery that it'll run afoul of evaluation rules